File: app/src/test/java/bj/discogsbrowser/search/SearchFactory.java
package bj.discogsbrowser.search;

import java.util.List;

import bj.discogsbrowser.greendao.SearchTerm;
import bj.discogsbrowser.model.search.SearchResult;
import edu.emory.mathcs.backport.java.util.Arrays;

/**
 * Created by Josh Laird on 13/05/2017.
 */

public class SearchFactory
{
    public SearchTerm getPastSearch()
    {
        SearchTerm searchTerm = new SearchTerm();
        searchTerm.setSearchTerm("yeson");
        return searchTerm;
    }

    public List<SearchResult> getThreeEmptySearchResults()
    {
        SearchResult[] searchResults = {buildSearchResult(), buildSearchResult(), buildSearchResult()};
        return Arrays.asList(searchResults);
    }

    public List<SearchResult> getArtistAndReleaseSearchResult()
    {
        SearchResult[] searchResults = {buildArtistSearchResult(), buildReleaseSearchResult()};
        return Arrays.asList(searchResults);
    }


    public SearchResult buildArtistSearchResult()
    {
        SearchResult searchResult = new SearchResult();
        searchResult.setTitle("ye");
        searchResult.setType("artist");
        searchResult.setThumb("yeeee");
        return searchResult;
    }

    public SearchResult buildReleaseSearchResult()
    {
        SearchResult searchResult = new SearchResult();
        searchResult.setTitle("ye");
        searchResult.setType("release");
        searchResult.setThumb("yeeee");
        return searchResult;
    }

    public SearchResult buildSearchResult()
    {
        SearchResult searchResult = new SearchResult();
        searchResult.setTitle("ye");
        searchResult.setType("yeee");
        searchResult.setThumb("yeeee");
        return searchResult;
    }
}

File: app/src/testShared/bj/vinylbrowser/search/SearchFactory.kt
package bj.vinylbrowser.search

import bj.vinylbrowser.greendao.SearchTerm
import bj.vinylbrowser.model.search.SearchResult

/**
 * Created by Josh Laird on 21/05/2017.
 */
object SearchFactory {
    // Java seems to interpret all of these SearchTerms as the same when using .indexOf() so this
    // must be incremented to be recognised as different
    var index: Int = 0

    @JvmStatic fun buildThreeSearchResults(): List<SearchResult> {
        return listOf(buildSearchResult("release"), buildSearchResult("release"),
                buildSearchResult("release"))
    }

    @JvmStatic fun buildArtistAndReleaseSearchResult(): List<SearchResult> {
        return listOf(buildSearchResult("artist"), buildSearchResult("release"))
    }

    @JvmStatic fun buildPastSearch(): SearchTerm {
        val searchTerm = SearchTerm()
        searchTerm.searchTerm = "yeson"
        return searchTerm
    }

    private fun buildSearchResult(type: String): SearchResult {
        val result = SearchResult()
        result.id = index++.toString()
        result.title = "title"
        result.type = type
        result.thumb = "thumb"
        return result
    }
}